### PR TITLE
Merge `delphi_web_python` Docker image from `operations` repo

### PR DESCRIPTION
**Pull request by dshemetov**
This is part of #965.

Having our Dockerfiles be partially compiled from the `operations` repo and partially here introduces unnecessary complexity, with no clear benefits (that I can see).

In the JIT work, I need to bump the pandas version, so moving the Python `requirements.txt` file from the `operations` repo here lets me do that in just one place.

I believe this is the last of our three Docker images to depend on the operations repo:
- `delphi_database_epidata` is already fully built from `dev/docker/database/epidata/Dockerfile` in this repo
- `delphi_web_epidata` is also fully built from `devops/Dockerfile` in this repo

This doesn't fully decouple us from `operations` as the `setup.sh` script still copies many old repos to the `delphi_web_python` image. We can consider excising those in a different PR.

**Prerequisites**:

- [x] Unless it is a documentation hotfix it should be merged against the `dev` branch
- [x] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [x] Build is successful
- [x] Code is cleaned up and formatted

### Summary

This PR:
- merged the [Dockerfile for the `delphi_python` image](https://github.com/cmu-delphi/operations/tree/main/dev/docker/python) into the Dockerfile for the `delphi_web_python` image in this repo
- copied the Python `requirements.txt` file from [that repo](https://github.com/cmu-delphi/operations/tree/main/dev/docker/python/assets) and merged it in with `requirements.dev.txt` 
- copied `setup.sh` file from [that repo](https://github.com/cmu-delphi/operations/tree/main/dev/docker/python/assets) to `dev/docker/python` in this repo
- renamed `requirements.txt` to `requirements.api.txt` and sorted dependencies
- deduplicated packages between `requirements.api.txt` and `requirements.dev.txt` and removed some unused ones (like `scikit-learn` from `requirements.dev.txt`)
- pinned package versions in `requirements.dev.txt` with the latest working versions

### CODE REVIEW

**Comment by melange396** [comment]
nice!  you can then remove these lines as well: https://github.com/cmu-delphi/delphi-epidata/blob/3d37c593ce460ec0bce49f28eafa2284935d0c94/dev/local/Makefile#L124-L125

**Comment by melange396** [comment]
and looks like we couldve/shouldve removed these lines already too:
https://github.com/cmu-delphi/delphi-epidata/blob/3d37c593ce460ec0bce49f28eafa2284935d0c94/dev/local/Makefile#L101-102

File: dev/docker/python/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,22 @@
 # start with the `delphi_python` image
```
Comment (melange396) [inline dev/docker/python/Dockerfile:1]:
```suggestion
```

**Comment by melange396** [comment]
with this PR, we have 3 different "requirements.txt" files in this repo: (ignoring `./src/server/covidcast_issues_migration/requirements.txt` from an old one-off thing)
- `./dev/docker/python/requirements.txt`  (which you just added/copied here)
- `./requirements.txt`
- `./requirements.dev.txt`

all 3 are used in `dev/docker/python/Dockerfile` to create `delphi_web_python`, which is only really used for tests.

only `./requirements.txt` (in this repo's root dir) is used to create `delphi_web_epidata` which runs the api server.

i have a PR (#1046) against this branch.  my changes sort the 3 requirements files, and remove duplicated entries from the new one.  we should prune down that file some more if we can, maybe even remove it entirely by moving its entries to the other two files.

a caveat is that `.github/workflows/update_gdocs_data.yml` also uses `requirements.dev.txt`, and im not sure what it does -- the meat of it seems to be in its line 26 "run: inv update-gdoc" but i cant figure out what that does

**Comment by dshemetov** [comment]
`update_gdocs_data.yml` updates the local `db_sources.csv` and `db_signals.csv` files (that the server uses to understand what signals are available) from [this Google sheet](https://docs.google.com/spreadsheets/d/e/2PACX-1vRfXo-qePhrYGAoZqewVnS1kt9tfnUTLgtkV7a-1q7yg4FoZk0NNGuB1H6k10ah1Xz5B8l1S1RB17N6/pub?gid=329338228) (and another one elsewhere).

**Comment by dshemetov** [comment]
Merged in George's mods from #1046.

**Comment by melange396** [comment]
looks pretty good to me:
- removes a cross-repo dependency
- should speed up our build process (including ci)
- generally reduces complexity

my concerns that remain:
- looks like `xlrd` and `matplotlib` can come out of the requirements file(s)?
- do we wanna add `scipy` back into the one requirements file?  our one reference to it is probably being taken care of by a chained dependency, so its not a huge deal
- im not sure we need or want to pin so many of those version numbers, but that can stand for the moment as i am soon going to create a github issue about version pinning
- we may want to "mark for deletion" a number of the things in the `operations` repo that we dont need anymore (`Dockerfile`, `setup.sh`, `requirements.txt`, etc...)

**Review by melange396 — DISMISSED** [review]

**Review by melange396 — APPROVED** [review]

**Pull request merged**